.circleci/config.yml

```diff
@@ -112,11 +112,52 @@ jobs:
             include_visit_job_action: false
             failure_message: '${SLACK_FAILURE_MESSAGE}'
             success_message: '${SLACK_SUCCESS_MESSAGE}'
+  UI Tests:
+    # Props to work done by others in this file and in
+    # https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/.circleci/config.yml
+    # this is a combined rip-off from both
+    parameters:     
+      device:
+        description: The device (e.g. "iPhone 12") to use when running tests.
+        type: string
+        default: "iPhone 12"
+    executor:
+      name: ios/default
+      xcode-version: "12.4.0"
+    environment:
+      FL_OUTPUT_DIR: output
+      FASTLANE_LANE: run_UI_tests
+    steps:
+      - git/shallow-checkout
+      - ios/install-dependencies:
+            bundle-install: true
+            pod-install: true
+      - run:
+          name: Copy Secrets
+          command: bundle exec fastlane run configure_apply
+      # Start: Swift Package Manager Workaround
+      # Ref. https://support.circleci.com/hc/en-us/articles/360044709573-Swift-Package-Manager-fails-to-clone-from-private-Git-repositories
+      - run: rm ~/.ssh/id_rsa
+      - run: for ip in $(dig @8.8.8.8 github.com +short); do ssh-keyscan github.com,$ip; ssh-keyscan $ip; done 2>/dev/null >> ~/.ssh/known_hosts || true
+      # END: Swift Package Manager Workaround
+      - run:
+          name: Fastlane
+          command: bundle exec fastlane $FASTLANE_LANE device:'<< parameters.device >>'
+      - store_artifacts:
+          path: output
+      - store_test_results:
+          path: output/scan
 
 workflows:
   simplenote_ios:
     jobs:
       - Test
+      - UI Tests:
+          name: UI Tests (iPhone 12)
+          device: "iPhone 12"
+          filters:
+            branches:
+              only: /.*UI-tests.*/
```

Comment:
You asked about branch filter vs cron schedule vs other options in the PR description.

What do you think is the best option? I think the answer also depends on how soon we want to know if the tests break: as soon as code changes? before every release? ad hoc? once a day?

Reply:
@mokagio I'm thinking of keeping it as it is now, for next week 🤔 

I've applied everything you've mentioned in review, [apart](https://github.com/Automattic/simplenote-ios/pull/1225#discussion_r610572013) from adding `deployment_target_version` 🙇 Thank you! After limiting the copied artefacts to `report.junit` and `report.html` only, the job takes 2 minutes less.

Next week (Apr 12-18) I'll need to wrap up with automation of `Settings` tests, which is a final part of my project. During this I'm going to use branches that contain `UI-tests` part and will see how it's going to work.

Reply:
> What do you think is the best option?

Generally, in ideal case, I would like to see the tests run as much a possible, on every change. We have them now, and it costs us nothing but time. But the more tests will we select, the more problematic will it be. Probably, it will not be suitable for everyone.

I'm for having two options.

1. A subset of tests, the rule to run them can be discussed with the team.
and
2. Running all tests, but with approval.

I'm thinking to ask in Simpelnote P2 next week about 1 and gather opinions.

Reply:
> Generally, in ideal case, I would like to see the tests run as much a possible, on every change

😄 

> We have them now, and it costs us nothing but time

Worth pointing out that CI time does cost money, too. Ideally, money spent in CI means less money lost as a result of users experiencing bugs and us doing creative rather than rote work, but it's good to remember that too.

As you say, the more tests we'll have the longer they'll take to run. In that case, the feedback time grows and it might become more effective to run them ad hoc and wait for feedback or use a scheduler.

Keen to see what comes out of the P2 discussion. I think this project is the ideal place where to experiment with different UI tests integration in CI workflows 👍

Reply:
> Worth pointing out that CI time does cost money, too.

TIL. I didn't know we pay for CI time. For some reason, I thought this is a subscription for a number of jobs to run in parallel.

Reply:
🤔 now that you mention it... to be honest, I'm not totally sure what our arrangement is with CircleCI. If not a matter of pay-by-the-minute it's a matter of parallelization and memory available in the hardware we use. If we were to run so many tests to bloat the queue, we'd then end up paying for implicitly in terms of delayed feedback, or explicitly by throwing more money at the problem and getting more or faster hardware to reduce the time in the queue.